gnovm/pkg/gnolang/machine.go

```diff
@@ -2131,25 +2131,25 @@ func (m *Machine) String() string {
 		totalLength = vsLength + ssLength + xsLength + bsLength + obsLength + fsLength + exceptionsLength
 	)
 
-	var builder strings.Builder
+	builder := new(strings.Builder)
 	builder.Grow(totalLength)
 
-	builder.WriteString(fmt.Sprintf("Machine:\n    PreprocessorMode: %v\n    Op: %v\n    Values: (len: %d)\n", m.PreprocessorMode, m.Ops[:m.NumOps], m.NumValues))
+	fmt.Fprintf(builder, "Machine:\n    PreprocessorMode: %v\n    Op: %v\n    Values: (len: %d)\n", m.PreprocessorMode, m.Ops[:m.NumOps], m.NumValues)
```

Comment:
How about explicitly use the builder's pointer by passing `&builder` in `fmt.Fprintf`? Of course, the compiler will handle it implicitly.

```suggestion
	fmt.Fprintf(&builder, "Machine:\n    PreprocessorMode: %v\n    Op: %v\n    Values: (len: %d)\n", m.PreprocessorMode, m.Ops[:m.NumOps], m.NumValues)
```

Reply:
I went with the much faster and better variant of this which still uses a pointer but takes it only once
```diff
var sb strings.Builder
builder := &sb
```

and has a reduction in CPU time consumption by 1% more.